File: appfw/src/main/java/com/jx/appfw/data/repository/MainDataRepository.java
package com.jx.appfw.data.repository;

import com.jx.appfw.data.repository.datasource.MainDataStore;
import com.jx.appfw.data.repository.datasource.MainDataStoreFactory;
import com.jx.appfw.domain.repository.MainRepository;
import com.jx.appfw.domain.request.main.TestBean;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Observable;

/**
 * {@link MainRepository} for retrieving menu.
 */
@Singleton
public class MainDataRepository implements MainRepository {
    private static String TAG = MainDataRepository.class.getName();

    private final MainDataStoreFactory mainDataStoreFactory;

    private final MainDataStore mainNetDataStore;

    private final MainDataStore mainDataStore;

    private final MainDataStore mainCacheDataStore;

    @Inject
    MainDataRepository(MainDataStoreFactory mainDataStoreFactory) {
        this.mainDataStoreFactory = mainDataStoreFactory;
        this.mainDataStore = mainDataStoreFactory.createMainDataStore();
        this.mainNetDataStore = mainDataStoreFactory.createNetMainDataStore();
        this.mainCacheDataStore = mainDataStoreFactory.createCacheMainDataStore();
    }

    @Override
    public Observable<String> test(TestBean testBean) {
        return this.mainNetDataStore.test(testBean);
    }
}

File: appfw/src/main/java/com/jx/appfw/data/repository/MainDataRepository.kt
package com.jx.appfw.data.repository

import com.jx.appfw.data.repository.datasource.MainDataStore
import com.jx.appfw.data.repository.datasource.MainDataStoreFactory
import com.jx.appfw.domain.repository.MainRepository
import com.jx.appfw.domain.request.main.TestBean
import io.reactivex.Observable
import javax.inject.Inject
import javax.inject.Singleton

class MainDataRepository(val mMainDataStoreFactory: MainDataStoreFactory) : MainRepository {
    private val mainNetDataStore: MainDataStore
    private val mainCacheDataStore: MainDataStore

    init {
        mainNetDataStore = mMainDataStoreFactory.createNetMainDataStore()
        mainCacheDataStore = mMainDataStoreFactory.createCacheMainDataStore()
    }

    override fun test(testBean: TestBean): Observable<String> {
        return this.mainNetDataStore.test(testBean)
    }
}